src/colab_gitlab/signals.py

```diff
@@ -0,0 +1,105 @@
+
+import logging
+
+import requests
+
+from django.conf import settings
+from django.dispatch import receiver
+
+from colab.accounts.signals import user_password_changed, user_created
+
+LOGGER = logging.getLogger('colab.plugins.gitlab')
+
+
+@receiver(user_password_changed)
+def update_gitlab_password(sender, **kwargs):
+    user = kwargs.get('user')
+    password = kwargs.get('password')
+
+    app_config = settings.COLAB_APPS.get('colab_gitlab', {})
+    private_token = app_config.get('private_token')
+    upstream = app_config.get('upstream', '').rstrip('/')
+
+    error_msg = u'Error trying to update "%s" password on Gitlab. Reason: %s'
+
+    users_endpoint = '{}/api/v3/users'.format(upstream)
+
+    params = {'search': user.username, 'private_token': private_token}
+    try:
+        response = requests.get(users_endpoint, params=params)
```

Comment:
Our current production server does not have a valid ssl_certificate. I tested this using cdtc server and got the response.
Reason: Request to API failed ([Errno 1] _ssl.c:504: error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed)
To fix this temporarily I used verify=False 
We could leave verify=False for now.